### PR TITLE
Add logic to focus first web component in array

### PR DESCRIPTION
**Pull request by rhasselle-oddball**
## Summary

- Allow web components to be included when trying to focus the first element of arrays

## Related issue(s)

- _Link to ticket created in va.gov-team repo_
department-of-veterans-affairs/va.gov-team#62008

## Testing done

- Browser and unit testing

## What areas of the site does it impact?

All forms that use a arrays with web components

### Quality Assurance & Testing

- [ ] I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [ ] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [ ] Linting warnings have been addressed
- [ ] Documentation has been updated ([link to documentation](#) \*if necessary)
- [ ] Screenshot of the developed feature is added
- [ ] [Accessibility testing](https://depo-platform-documentation.scrollhelp.site/developer-docs/wcag-2-1-success-criteria-and-foundational-testing) has been performed

### Error Handling

- [ ] Browser console contains no warnings or errors.
- [ ] Events are being sent to the appropriate logging solution
- [ ] Feature/bug has a monitor built into Datadog or Grafana (if applicable)

### Authentication

- [ ] Did you login to a local build and verify all authenticated routes work as expected with a test user

### :warning: Team Sites (only applies to modifications made to the VA.gov header) :warning:

- [ ] The vets-website header does not contain any web-components
- [ ] I used the [proxy-rewrite steps](https://github.com/department-of-veterans-affairs/vets-website/tree/main/src/applications/proxy-rewrite#local-dev) to test the injected header scenario
- [ ] I reached out in the `#sitewide-public-websites` Slack channel for questions

## Requested Feedback

(OPTIONAL) _What should the reviewers know in addition to the above. Is there anything specific you wish the reviewer to assist with. Do you have any concerns with this PR, why?_

### CODE REVIEW

File: src/platform/forms-system/src/js/fields/ArrayField.jsx
Comment (rhasselle-oddball) [inline src/platform/forms-system/src/js/fields/ArrayField.jsx:290]: I couldn't think of any good reasoning why tagName would be targeted here explicitly. It seems like we would just want to focus the first element. and tagName doesn't always play nice with web components.

**Review by powellkerry — APPROVED** [review]

**Comment by rhasselle-oddball** [comment]
There are some "field disabled" Cypress 11 bugs that relate to web components. I think this is blocked by needing Cypress 12. Related PR that would unblock this: https://github.com/department-of-veterans-affairs/vets-website/pull/25149

**Pull request merged**
